e2e/testcases/apiservice_test.go

```diff
@@ -52,36 +53,48 @@ func TestCreateAPIServiceAndEndpointInTheSameCommit(t *testing.T) {
 		nt.T.Fatalf("validate failed %s", err)
 	}
 
+	// Test done, removing the test APIService first to prevent Discovery failure blocking
+	// the test repo from cleaning up
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/cluster/apiservice.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter API service")
 	nt.WaitForRepoSyncs()
 
+	// Remove the backend Deployment of test APIService
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace-custom-metrics.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter deployment")
-	nt.WaitForRepoSyncs()
-
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter namespace")
 	nt.WaitForRepoSyncs()
 }
 
 func TestImporterAndSyncerResilientToFlakyAPIService(t *testing.T) {
 	nt := nomostest.New(t, ntopts.RequireGKE(t), ntopts.Unstructured)
-	nt.T.Log("Adding API service with backend temporarily unavailable")
-	nt.MustKubectl("apply", "-f", "../testdata/apiservice/apiservice.yaml")
 	nt.T.Cleanup(func() {
 		nt.MustKubectl("delete", "-f", "../testdata/apiservice/apiservice.yaml", "--ignore-not-found")
+		nt.MustKubectl("delete", "-f", "../testdata/apiservice/namespace-custom-metrics.yaml", "--ignore-not-found")
+		nt.MustKubectl("delete", "-f", "../testdata/apiservice/namespace.yaml", "--ignore-not-found")
+		nt.MustKubectl("delete", "-f", "../testdata/apiservice/rbac.yaml", "--ignore-not-found")
+
 		if t.Failed() {
 			nt.PodLogs(adapterNamespace, adapterName, "pod-custom-metrics-stackdriver-adapter", true)
 		}
 	})
 
-	nt.T.Log("Creating commit with resources")
-	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/rbac.yaml", "acme/cluster/rbac.yaml")
+	nt.T.Log("Adding APIService with backend temporarily unavailable to trigger discovery errors")
+	nt.MustKubectl("apply", "-f", "../testdata/apiservice/apiservice.yaml")
+
+	nt.T.Log("Creating commit with test resources")
+	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/namespace-resilient.yaml", "acme/namespaces/resilient/namespace.yaml")
+	//nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/apiservice.yaml", "acme/cluster/apiservice.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("add testing resources")
 
-	nt.T.Log("Adding backend")
+	nt.T.Log("Waiting for APIServerError to be present in Rootsync status")
+	if !nt.MultiRepo {
+		nt.WaitForRepoSourceError(status.APIServerErrorCode)
+	}
```

Comment:
else?

The test runs on both mono-repo mode and multi-repo mode.

Reply:
Just confirmed in debugging env that under multirepo mode the test resource can be correctly reconciled even with bad APIService. Added RG status check for the test resource to make sure.